### DISCUSSION TITLE
Render multiple mdx file into a single page

### DISCUSSION THREAD

**Discussion by openscript** [gatsbyjs/gatsby#37301] [Help]
Hello, what is the way to go, if you want to render multiple markdown files (fragments, blocks, ..) into a single page with `gatsby-plugin-mdx`? In version 3 we had `<MDXRenderer>` but now it seems like this plugin is only usable for whole mdx pages? What is the way to go now?  I've found similar questions here https://stackoverflow.com/questions/74309351/how-to-render-mdx-with-gatsby-plugin-mdx-from-allmdx-query and https://github.com/gatsbyjs/gatsby/discussions/37045

![image](https://user-images.githubusercontent.com/1105080/208671499-31d95157-ced7-44b5-a4ec-031b91e5971a.png)

**Comment by rburgst** [comment]
had this question here myself https://github.com/gatsbyjs/gatsby/discussions/25068#discussioncomment-4420000 does not look as if there is any movement on that front

**Comment by openscript** [comment]
We had no other choice than to mix https://www.gatsbyjs.com/plugins/gatsby-transformer-remark/ and https://www.gatsbyjs.com/plugins/gatsby-plugin-mdx

**Reply by rburgst** [reply]
Can you explain a bit more? I am still at gatsby 4.x as I need that feature

**Reply by openscript** [reply]
In our case we could use `gatsby-transformer-remark` for Markdown content and `gatsby-plugin-mdx` for pages, which also load the Markdown content via `gatsby-transformer-remark`, but I think that doesn't make sense in all cases.

**Reply by pjuerg** [reply]
I also have this pattern with various MDX queries per page. Any news on how to replace the MDX renderer?

**Reply by openscript** [reply]
Unfortunately no.

**Reply by Papershine** [reply]
Don't know if this applies in your case, but I wanted to use my React components in my markdown files (with multiple markdown files in a single page). Since `MDXRenderer` didn't seem to work anymore, I instead tried to use `rehype-react` on top of `gatsby-transformer-remark` to hydrate custom HTML tags with the React component and it seems to be working. I followed the guide at [https://using-remark.gatsbyjs.org/custom-components/](https://using-remark.gatsbyjs.org/custom-components) (for future reference).

**Comment by mafritsch** [comment]
It's 2024 and I still find the same problem. I used to render single MDX files on a  What's New page using MDXRenderer with Gatsby 4. Now, I want to upgrade to Gatsby 5 due to security reasons. However, MDXRenderer is gone in gatsby-plugin-mdx. I tried the compile method from mdx-js/react but without success. Is there a know way to render single MDX files on  a Gatsby page?
